Trim reqs

kquick opened the pull request:
The original set of requirements is moved  to requirements_full.txt, and requirements.txt is reduced to the minimum set of requirements needed for executing the direct code in this package.

Original:
  requirements: 116
  pip install time: 1m30s
  installed size: 714 MB  (plus system package requirements)

Updated:
  requirements: 3
  pip install time: 8.7s
  installed size: 160MB

tuxji reviewed (commented):
@kquick Your changes look great, but requirements.txt has conflicts due to more recent changes in the main branch.  Probably the quickest way (if it works) to fix these conflicts is to pull the latest upstream changes in your main branch, save a copy of its latest requirements.txt somewhere, run git reset --soft main in your trim_reqs branch (this resets your branch to the latest upstream changes while leaving your changed files as locally modified files), replace requirements_full.txt with your saved copy, commit your changed files again, and git push --force to update your pull request.

kquick commented:
My usual preference is to have the history for forensics when addressing issues, and I didn't see any documentation about preferences for this project so that was my default approach, but you should be able to squash merge if you don't want the full history.
